# Settings for ros_control control loop
generic_hw_control_loop:
  loop_hz: 20
  cycle_time_error_threshold: 0.5

# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - arm_1_joint
    - arm_2_joint
    - arm_3_joint
    - arm_4_joint
    - arm_5_joint
  sim_control_mode: 0 # 0: position, 1: velocity

  ip_address: 10.4.20.180
  control_port: 10100
  status_port: 10000
  profile_no: 2
  speed: 100
  speed2: 0
  accel: 100
  decel: 100
  accel_ramp: 0.1
  decel_ramp: 0.1
  in_range: -1
  straight: 0 #0=rotational; -1=straight

  post_init_configuration: [0.085, -1.552, 3.136, 3.155, 0.038] #config where the arm should move to after successful initialization


# Publish all joint states ----------------------------------
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Joint Trajectory Controller -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
joint_trajectory_controller:
  type: position_controllers/JointTrajectoryController
  # These joints can likely just be copied from the hardware_interface list above
  joints:
    - arm_1_joint
    - arm_2_joint
    - arm_3_joint
    - arm_4_joint
    - arm_5_joint
  constraints:
    goal_time: 10.0
    #stopped_position_tolerance: 0.4 # Defaults to 0.01
    arm_1_joint:
      trajectory: 10.0
      goal: 0.005
    arm_2_joint:
      trajectory: 10.0
      goal: 0.01
    arm_3_joint:
      trajectory: 10.0
      goal: 0.01
    arm_4_joint:
      trajectory: 10.0
      goal: 0.01
    arm_5_joint:
      trajectory: 10.0
      goal: 0.005
  # state_publish_rate:  50 # Defaults to 50
  # action_monitor_rate: 20 # Defaults to 20
  #hold_trajectory_duration: 0 # Defaults to 0.5

# Individual Position Controllers ---------------------------------------
# Allows to send individual ROS msg of Float64 to each joint separately
arm_1_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: arm_1_joint
arm_2_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: arm_2_joint
arm_3_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: arm_3_joint
arm_4_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: arm_4_joint
arm_5_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: arm_5_joint


# Group Position Controllers ---------------------------------------
# Allows to send single ROS msg of Float64MultiArray to all joints
joint_position_controller:
  type: position_controllers/JointGroupPositionController
  joints:
    - arm_1_joint
    - arm_2_joint
    - arm_3_joint
    - arm_4_joint
    - arm_5_joint
